base_layer/wallet/src/output_manager_service/service.rs

```diff
@@ -67,20 +71,34 @@ where T: OutputManagerBackend
             OutputManagerRequest,
             Result<OutputManagerResponse, OutputManagerError>,
         >,
-        master_key: PrivateKey,
-        branch_seed: String,
-        primary_key_index: usize,
+        config: OutputManagerConfig,
         db: OutputManagerDatabase<T>,
-    ) -> OutputManagerService<T>
+    ) -> Result<OutputManagerService<T>, OutputManagerError>
     {
-        OutputManagerService {
-            key_manager: Mutex::new(KeyManager::<PrivateKey, KeyDigest>::from(
-                master_key,
-                branch_seed,
-                primary_key_index,
-            )),
-            db,
-            request_stream: Some(request_stream),
+        if config.master_key.is_none() && config.seed_words.is_none() {
+            return Err(OutputManagerError::InvalidConfig);
+        }
+
+        if config.master_key.is_some() {
```

Comment:
nit: `match` statement would mean you wouldn't need to have an error if `config.master_key` is None (which is not possible in the Some branch)